cognite/client/data_classes/contextualization.py

```diff
@@ -1036,3 +1038,213 @@ def __init__(
+            ... )
+    """
+
+    def __init__(
+        self,
+        annotation_extract: bool | None = None,
+        case_sensitive: bool | None = None,
+        connection_flags: ConnectionFlags | list[str] | None = None,
+        customize_fuzziness: CustomizeFuzziness | dict[str, Any] | None = None,
+        direction_delta: float | None = None,
+        direction_weights: DirectionWeights | dict[str, Any] | None = None,
+        min_fuzzy_score: float | None = None,
+        read_embedded_text: bool | None = None,
+        remove_leading_zeros: bool | None = None,
+        substitutions: dict[str, list[str]] | None = None,
+        **params: Any,
+    ) -> None:
+        self.annotation_extract = annotation_extract
+        self.case_sensitive = case_sensitive
+        self.connection_flags = connection_flags
+        self.customize_fuzziness = customize_fuzziness
+        self.direction_delta = direction_delta
+        self.direction_weights = direction_weights
+        self.min_fuzzy_score = min_fuzzy_score
+        self.read_embedded_text = read_embedded_text
+        self.remove_leading_zeros = remove_leading_zeros
+        self.substitutions = substitutions
+        self._extra_params = {}
+
+        _known_params = {to_camel_case(k): k for k in vars(self)}
+        for param_name, value in params.items():
+            if known := _known_params.get(to_camel_case(param_name)):
+                raise ValueError(
+                    f"Provided parameter name `{param_name}` collides with a known parameter `{known}`. Please use it insead."
+                )
+            else:
+                self._extra_params[param_name] = value
+
+    def dump(self, camel_case: bool = True) -> dict[str, Any]:
+        dumped = super().dump(camel_case=camel_case)
+
+        for param_name, v in self._extra_params.items():
+            dumped[to_camel_case(param_name) if camel_case else param_name] = v
+
+        remove_keys: list[str] = []
+        for param_name, v in dumped.items():
+            if isinstance(v, CogniteObject):
+                dumped[param_name] = v.dump(camel_case=camel_case)
+            elif isinstance(v, ConnectionFlags):
+                cf = v.dump()
+                if cf:
+                    dumped[param_name] = cf
+                else:
+                    remove_keys.append(param_name)
+
+        for param_name in remove_keys:
+            del dumped[param_name]
+
+        return dumped
+
+    @classmethod
+    def _load(cls, resource: dict[str, Any], cognite_client: CogniteClient | None = None) -> Self:
+        resouce_copy = resource.copy()
+
+        if con_flg := resouce_copy.pop("connectionFlags", None):
+            con_flg = ConnectionFlags.load(con_flg)
+        if cus_fuz := resouce_copy.pop("customizeFuzziness", None):
+            cus_fuz = CustomizeFuzziness.load(cus_fuz)
+        if dir_wgt := resouce_copy.pop("directionWeights", None):
+            dir_wgt = DirectionWeights.load(dir_wgt)
```

Comment:
Perhaps we can change `pop` (mutates) to `get` (and skip initial copy)?
```suggestion
        if con_flg := resouce_copy.get("connectionFlags"):
            con_flg = ConnectionFlags.load(con_flg)
        if cus_fuz := resouce_copy.pop("customizeFuzziness"):
            cus_fuz = CustomizeFuzziness.load(cus_fuz)
        if dir_wgt := resouce_copy.pop("directionWeights"):
            dir_wgt = DirectionWeights.load(dir_wgt)
```